.github/workflows/typescript-all-files-check.yml

```diff
@@ -12,6 +12,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          path: currentBranch
```

Comment:
The details of why this was failing aren't 100% clear to me, but the basic issue was that we were checking out the target branch (e.g., staging) as a child directory of the root (where we checked out the current branch), and this led to some sort of dependency conflict with the parent directory. I have modified this GH action and `checkAllFilesForTypeErrorCount.ts` to use separate, non-nested paths. This is similar to how our 
"Typescript Reduce Type Errors" check works.